include/mata/nfa.hh

```diff
@@ -1251,49 +1235,81 @@ private:
 
 public:
-
-    EnumAlphabet() : symbol_map() { }
-
-    EnumAlphabet(const EnumAlphabet& rhs) : symbol_map(rhs.symbol_map) {}
-
     /**
      * Create alphabet from variable number of NFAs.
-     * @tparam Nfas Type Nfa.
-     * @param nfas NFAs to create alphabet from.
-     * @return Alphabet.
+     * @tparam[in] Nfas Type Nfa.
+     * @param[in] nfas NFAs to create alphabet from.
+     * @return Created alphabet.
      */
     template<typename... Nfas, typename = AreAllNfas<Nfas...>>
     static EnumAlphabet from_nfas(const Nfas&... nfas) {
         EnumAlphabet alphabet{};
+        // TODO: When we are on C++17, we can use fold expression here instead of the manual for_each_argument reimplementation.
         for_each_argument([&alphabet](const Nfa& aut) {
             size_t aut_num_of_states{ aut.get_num_of_states() };
             for (State state{ 0 }; state < aut_num_of_states; ++state) {
                 for (const auto& state_transitions: aut.transitionrelation[state]) {
-                    alphabet.add_symbol(std::to_string(state_transitions.symbol), state_transitions.symbol);
+                    alphabet.try_add_new_symbol(std::to_string(state_transitions.symbol), state_transitions.symbol);
                 }
             }
         }, nfas...);
         return alphabet;
     }
 
-    template <class InputIt>
-    EnumAlphabet(InputIt first, InputIt last) : EnumAlphabet()
-    { // {{{
-        size_t cnt = 0;
-        for (; first != last; ++first)
-        {
-            bool inserted;
-            std::tie(std::ignore, inserted) = symbol_map.insert({*first, cnt++});
-            if (!inserted)
-            {
-                throw std::runtime_error("multiple occurrence of the same symbol");
+    /**
+     * Create alphabet from vector of of NFAs.
+     * @param[in] nfas Vector of NFAs to create alphabet from.
+     * @return Created alphabet.
+     */
+    static EnumAlphabet from_nfas(const ConstAutRefSequence& nfas) {
+        EnumAlphabet alphabet{};
+        size_t nfa_num_of_states{};
+        for (const auto& nfa: nfas) {
+            nfa_num_of_states = nfa.get().get_num_of_states();
+            for (State state{ 0 }; state < nfa_num_of_states; ++state) {
+                for (const auto& state_transitions: nfa.get().transitionrelation[state]) {
+                    alphabet.update_next_symbol_value(state_transitions.symbol);
+                    alphabet.try_add_new_symbol(std::to_string(state_transitions.symbol), state_transitions.symbol);
+                }
             }
         }
-    } // }}}
+        return alphabet;
+    }
 
-    EnumAlphabet(std::initializer_list<std::string> l) :
-            EnumAlphabet(l.begin(), l.end())
-    { }
+    /**
+     * Create alphabet from vector of of NFAs.
+     * @param[in] nfas Vector of pointers to NFAs to create alphabet from.
+     * @return Created alphabet.
+     */
+    static EnumAlphabet from_nfas(const std::vector<const Nfa*>& nfas) {
```

Comment:
Isn't this type named previously? Like `NfaSequence`?

Reply:
`NfaSequence` is `std::vector<Nfa>`. I want pointers to NFAs here. However, I will add a new types `ConstAutPtrSequence` and `AutPtrSequence` similar to the following:
```cpp
using AutRefSequence = RefSequence<Nfa>; ///< A sequence of references to non-deterministic finite automata.
using ConstAutRefSequence = RefSequence<const Nfa>; ///< A sequence of const references to non-deterministic finite automata.
```